funds-manager/funds-manager-server/src/execution_client/quotes.rs

```diff
@@ -45,19 +62,60 @@ impl ExecutionClient {
     /// Fetch a quote for an asset
     pub async fn get_quote(
         &self,
-        buy_asset: &str,
-        sell_asset: &str,
+        buy_asset: Address,
+        sell_asset: Address,
         amount: u128,
-        recipient: &str,
+        wallet: &LocalWallet,
     ) -> Result<ExecutionQuote, ExecutionClientError> {
+        // First, set an approval for the sell token, the 0x api will not give a quote
+        // if its contract is not an approved spender for the requested amount
+        let exchange_addr = Address::from_str(EXCHANGE_PROXY_ADDRESS).unwrap();
+        self.approve_erc20_allowance(sell_asset, exchange_addr, U256::from(amount), wallet).await?;
```

Comment:
my preference is still to err on the side of not having unused approvals on external contracts, which would imply setting the allowance back down to 0 after hitting the quote endpoint, and invoking another approval ahead of executing a swap (w/ revocation in the case of a failed swap). I think this is worth the gas cost & complexity, just another approval call. lmk what u think

Reply:
let's see how this goes and decide whether we want that. i really don't find covering all possible dangers on any margin of implied complexity to be a productive path.